### PR TITLE
HODS-407 | Return empty data set when no data found in IF

### PR DESCRIPTION
**Pull request by ajchilts**

### CODE REVIEW

File: app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala [hunk 1]
```diff
@@ -139,8 +143,14 @@ class IfConnector @Inject()(
       auditHelper.auditIfApiFailure(correlationId, matchId, request, requestUrl, msg)
       msg.contains("NO_DATA_FOUND") match {
         case true =>
-          logger.warn("Integration Framework NotFoundException encountered")
-          Future.failed(new NotFoundException(msg))
+          if (requestUrl.contains("counts"))
```
Comment (parkeradam) [inline app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala:146]: I dont really like the placement of this, why should the IF connector simply used for making calls to IF know about all the different URLs. I'd prefer to add this recoery code to the individuals services as they are already specific to each endpoint so makes sense to have these moved to each service. Also stops this growing as future devs add more and more endpoints.

**Review by parkeradam — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala (hunk 1 above)
Comment (ajchilts) [reply app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala:146]: @parkeradam  - The IF connector already knows about the different URLs as they are coded into the IF connector. The recover method is generic and applies to all endpoints. The empty objects being returned are IF objects and created via the IF calls. I will however explore alternative(s) and see what that might look like in code.

Comment (ajchilts) [reply app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala:146]: `requestUrl` is the IF request.

File: app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala [hunk 2]
```diff
@@ -43,6 +43,10 @@ class IfConnector @Inject()(
 
   private val baseUrl = servicesConfig.baseUrl("integration-framework")
 
+  val emptyEmployeeCountResponse = EmployeeCountResponse(None, None, Some(Seq()))
+  val emptyCtReturn              = CorporationTaxReturnDetailsResponse(None, None, None, Some(Seq()))
+  val emptySaReturn              = SelfAssessmentReturnDetailResponse(None, None, None, None, Some(Seq()))
```
Comment (parkeradam) [inline app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala:48]: Why a Some at the end and not a None? (both here and line 46)

File: app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala (hunk 1 above)
Comment (parkeradam) [reply app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala:146]: My bad, didnt see in the context of the whole file.

File: app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala (hunk 2 above)
Comment (ajchilts) [reply app/uk/gov/hmrc/organisationsdetailsapi/connectors/IfConnector.scala:48]: `None` gives no list where `some(Seq())` gives an empty list. An empty list looks better where there is no data.

**Review by parkeradam — APPROVED** [review]

**Pull request merged**